runner/test/cases/server/plugins/engine/nationalinsurancefield.test.ts

```diff
@@ -0,0 +1,81 @@
+import * as Code from "@hapi/code";
+import * as Lab from "@hapi/lab";
+import { NationalInsuranceField } from "server/plugins/engine/components/NationalInsuranceField";
+const lab = Lab.script();
+exports.lab = lab;
+const { expect } = Code;
+const { suite, describe, it } = lab;
+import sinon from "sinon";
+
+suite("NationalInsuranceField", () => {
+  describe("Generated schema", () => {
+    const componentDefinition: NationalInsuranceField = {
+      name: "NRBhkR",
+      options: {},
+      type: "NationalInsuranceField",
+      title: "title",
+      hint: "hint",
+      schema: {},
+      subtype: "",
+      list: null,
+      subType: "field",
+    };
+
+    const formModel = {
+      makePage: () => sinon.stub(),
+    };
+
+    const component = new NationalInsuranceField(
+      componentDefinition,
+      formModel
+    );
+    it("is required by default", () => {
+      console.log(component.formSchema.describe());
+      expect(component.formSchema.describe().flags.presence).to.equal(
+        "required"
+      );
+    });
+    it("is not required when explicitly configured", () => {
+      const component = new NationalInsuranceField(
+        {
+          ...componentDefinition,
+          options: { required: false },
+        },
+        formModel
+      );
+      expect(component.formSchema.validate("").error).to.not.exist();
+    });
+    it("validates NI number", () => {
+      let testCases: [any, boolean][] = [
+        ["AA123456C", true],
+        ["AA 12 34 56 C", true],
+        ["aa 12 34 56 c", true],
+        ["aa123456c", true],
+        ["", false],
+        [1234567890, false],
+        ["aa 12 34 56 c1", false],
+        ["aa123456c1", false],
+        ["aa123456", false],
+        ["AA123456C1", false],
+        ["AA123456", false],
```

Comment:
it looks like you're testing NIs that are
- in caps with no whitespace passes
- in caps with whitespaces passes
- in lowercase with whitespace
- in lowercase without whitespace
- empty 
- invalid format
- too long lowercase
- too short
- too long in caps
- too short in caps 

IMO, there doesn't need to be this many tests. not that it affects run time, but just because it means there's more to maintain. 

- in caps without whitespace
- in caps with whitespaces 
- in lowercase with whitespace
- in lowercase without whitespace <- you already have covered that lowercase, caps, whitespace, no whitespace
- empty 
- invalid format
- too long lowercase 
- too short lowercase  
- too long in caps <- not required, since you already have a case for too long
- too short in caps <- not required, since you have already have a case for too short


You should also consider splitting these out to `it("validates and passes"` (or something to that nature), and `it("validates and fails")` - then you don't have to do the if statement.

Reply:
Resolved.